spec.html

```diff
@@ -4524,10 +5441,18 @@ <h1>Abstract Equality Comparison</h1>
         1. If _x_ is *undefined* and _y_ is *null*, return *true*.
         1. If Type(_x_) is Number and Type(_y_) is String, return the result of the comparison _x_ == ! ToNumber(_y_).
         1. If Type(_x_) is String and Type(_y_) is Number, return the result of the comparison ! ToNumber(_x_) == _y_.
+        1. If Type(_x_) is BigInt and Type(_y_) is String, then
+          1. Let _n_ be StringToBigInt(_y_).
+          1. If _n_ is *NaN*, return *false*.
+          1. Return the result of the comparison _x_ == _n_.
+        1. If Type(_x_) is String and Type(_y_) is BigInt, return the result of the comparison _y_ == _x_.
         1. If Type(_x_) is Boolean, return the result of the comparison ! ToNumber(_x_) == _y_.
         1. If Type(_y_) is Boolean, return the result of the comparison _x_ == ! ToNumber(_y_).
-        1. If Type(_x_) is either String, Number, or Symbol and Type(_y_) is Object, return the result of the comparison _x_ == ToPrimitive(_y_).
-        1. If Type(_x_) is Object and Type(_y_) is either String, Number, or Symbol, return the result of the comparison ToPrimitive(_x_) == _y_.
+        1. If Type(_x_) is either String, Number, BigInt, or Symbol and Type(_y_) is Object, return the result of the comparison _x_ == ToPrimitive(_y_).
```

Comment:
do we not have an `IsPrimitiveType()` abstract operation to use here, and on the next line?

(we can add this in a followup PR)

Reply:
I don't think we have it, at least named as `IsPrimitiveType`. I think it is a good abstract operation to include, though.

Reply:
if it can be added prior to this PR, great; if it should be added as part of this PR, great; if after, great :-)

Reply:
Nice. I'll do in a follow up PR then.